frappe/core/doctype/doctype/test_doctype.py

```diff
@@ -686,6 +694,9 @@ def validate(code):
 		doctype.delete()
 		frappe.db.commit()
 
+	@unittest.skipUnless(
+		os.access(frappe.get_app_path("frappe"), os.W_OK), "Only run if frappe app paths is writable"
```

Comment:
frappe.get_app_path depends on `frappe.local` which is only present when site is `init`ed.

Reply:
convert this to a function call somehow? It shouldn't be ran when python module is being imported (we don't know if `frappe.local` is init-ted or not in that case.

Reply:
Oh, it looks like [it _is_ `init`ed](https://github.com/frappe/frappe/blob/3471685eaa046b3db805984f425be13fd9022f92/frappe/parallel_test_runner.py#L31), and it is [guaranteed to be](https://github.com/frappe/frappe/blob/3471685eaa046b3db805984f425be13fd9022f92/frappe/parallel_test_runner.py#L27-L28). Or have I missed something?

Reply:
It's a rather weak "guarantee", if someone imports this file in the test runner it will fail... similar instances have happened in the past :woozy_face: 


It's easy to fix, do something like this by wrapping unittest's skip decorator


 https://github.com/frappe/frappe/blob/84188587fa330a05b4c717a25cf1474e55757dd4/frappe/tests/test_query_builder.py#L24-L25

Reply:
welll, looks like this wrapping is also not truly "lazy" :pensive: